revolution/drivers.py

```diff
@@ -209,3 +211,334 @@ def voltages(self) -> dict[InputChannel, float]:
+    -Configurable parameters
+        -Full scale range of shunt voltage (±40.96mv or ±163.84mV)
+        -ADC conversion period (50us to 4120us)
+        -Number of averages (1 to 1024)
+        -External shunt calibrated resistance
+    """
+    
+    class ADC_range_enum(IntEnum):
+        LOW_PRECISION: bool = 0 #Shunt voltage range of ±163.84 mV
+        HIGH_PRECISION: bool = 1 #Shunt voltage range of ±40.96 mV
+
+    #Arguments to fill in
+    spi: SPI
+    voltage_step_down_ratio:float
+    shunt_resistance:float
+
+    _register_map = {
+        # name      :       [addr,         size (B),read/write,        signedness,      data]
+        "CONFIG":           {"addr":0x00, "size":2, "permission":"RW", "signed": False, "data":0x0000},
+        "ADC_CONFIG":       {"addr":0x01, "size":2, "permission":"RW", "signed": False, "data":0xFB68},
+        "SHUNT_CAL":        {"addr":0x02, "size":2, "permission":"RW", "signed": False, "data":0x1000},
+        "SHUNT_TEMPCO":     {"addr":0x03, "size":2, "permission":"RW", "signed": False, "data":0x0000},
+        "VSHUNT":           {"addr":0x04, "size":3, "permission":"RO", "signed": True, "data":0x000000},
+        "VBUS":             {"addr":0x05, "size":3, "permission":"RO", "signed": True, "data":0x000000},
+        "DIETEMP":          {"addr":0x06, "size":2, "permission":"RO", "signed": True, "data":0x0000},
+        "CURRENT":          {"addr":0x07, "size":3, "permission":"RO", "signed": True, "data":0x000000},
+        "POWER":            {"addr":0x08, "size":3, "permission":"RO", "signed": False, "data":0x000000},
+        "ENERGY":           {"addr":0x09, "size":5, "permission":"RO", "signed": False, "data":0x0000000000},
+        "CHARGE":           {"addr":0x0A, "size":5, "permission":"RO", "signed": True, "data":0x0000000000},
+        "DIAG_ALRT":        {"addr":0x0B, "size":2, "permission":"NA", "signed": False, "data":0x0001}, #This register has different read/write permissions per bit; must be handled individually
+        "SOVL":             {"addr":0x0C, "size":2, "permission":"RW", "signed": True, "data":0x7FFF},
+        "SUVL":             {"addr":0x0D, "size":2, "permission":"RW", "signed": True, "data":0x8000},
+        "BOVL":             {"addr":0x0E, "size":2, "permission":"RW", "signed": False, "data":0x7FFF},
+        "BUVL":             {"addr":0x0F, "size":2, "permission":"RW", "signed": False, "data":0x0000},
+        "TEMP_LIMIT":       {"addr":0x10, "size":2, "permission":"RW", "signed": True, "data":0x7FFF},
+        "PWR_LIMIT":        {"addr":0x11, "size":2, "permission":"RW", "signed": False, "data":0xFFFF},
+        "MANUFACTURER_ID":  {"addr":0x3E, "size":2, "permission":"RO", "signed": False, "data":0x5449},
+        "DEVICE_ID":        {"addr":0x3F, "size":2, "permission":"RO", "signed": False, "data":0x2291},
+    }
+    _adc_averaging_count_to_bin = bidict({1:0, 4:1, 16:2, 64:3, 128:4, 256:5, 512:6, 1024:7})
```

Comment:
Where does ``1`` map to? Probably ``0``, but is this intended? This causes confusion as the type ``bidict`` suggests ``1`` could also map to ``4``.